### PR TITLE
WINDUP-592  + WINDUP-595 Console UI WARNING if no "customer packages" specified.

### PR DESCRIPTION
**Pull request by OndraZizka**
Blocked by FORGE-2336 and affected by FORGE-2337.

### CODE REVIEW

File: config/api/src/main/java/org/jboss/windup/config/ValidationResult.java [hunk 1]
```diff
@@ -11,23 +11,30 @@
...
     {
-        this.success = success;
+        this.level = level;
         this.message = message;
     }
 
-    public boolean isSuccess()
+    public boolean isSucess()
+    {
+        return Level.ERROR.equals(level);
```
Comment (gastaldi) [inline config/api/src/main/java/org/jboss/windup/config/ValidationResult.java]: It is success if level == ERROR?

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/config/ScanPackagesOption.java [hunk 2]
```diff
@@ -51,6 +56,32 @@ public boolean isRequired()
...
+        }
+
         return ValidationResult.SUCCESS;
     }
+
+    private boolean isScanPackagesTooGeneral(List<String> includeJavaPackages)
+    {
+        if (includeJavaPackages != null)
+        {
+            Set<String> tooGeneral = new HashSet(Arrays.asList("com org net".split(" ")));
```
Comment (gastaldi) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/config/ScanPackagesOption.java]: Why create a HashSet here? You could simply use: 
````
List<String> tooGeneral = Arrays.asList("com","org","net");
````

File: config/api/src/main/java/org/jboss/windup/config/ValidationResult.java (hunk 1 above)
Comment (OndraZizka) [inline config/api/src/main/java/org/jboss/windup/config/ValidationResult.java]: Fixed

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/config/ScanPackagesOption.java (hunk 2 above)
Comment (OndraZizka) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/config/ScanPackagesOption.java]: A habit. Plus, I think, we might want to make it configurable in the future...

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/121/
Test PASSed.

**Comment by OndraZizka** [comment]
I've added WINDUP-595 to this since it also uses the ValidationResult changes.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/753/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/122/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/754/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/758/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/127/
Test PASSed.

**Comment by lincolnthree** [comment]
@OndraZizka Please advise when this can be reviewed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/762/<h2>Failed Tests: <span class='status-failure'>3</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/762/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/762/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/762/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>2</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/762/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/762/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/131/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/131/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/131/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/768/<h2>Failed Tests: <span class='status-failure'>3</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/768/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/768/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/768/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>2</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/768/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/768/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/137/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/137/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/137/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/769/<h2>Failed Tests: <span class='status-failure'>3</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/769/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/769/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/769/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>2</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/769/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/769/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/138/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/138/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/138/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by OndraZizka** [comment]
The test failure is not caused by this PR.

File: ui/addon/src/main/java/org/jboss/windup/ui/WindupCommand.java [hunk 3]
```diff
@@ -342,4 +364,25 @@ private boolean pathNotEmpty(File f)
         return false;
     }
 
+
+    private static class SimpleFailed implements Failed
```
Comment (lincolnthree) [inline ui/addon/src/main/java/org/jboss/windup/ui/WindupCommand.java]: Why not just use Results.fail("message")?

File: ui/addon/src/main/java/org/jboss/windup/ui/WindupCommand.java [hunk 4]
```diff
@@ -142,23 +148,38 @@ public void validate(UIValidationContext context)
...
 
     }
 
     @Override
-    public Result execute(UIExecutionContext context) throws Exception
+    public Result execute(UIExecutionContext uiExecContext) throws Exception
     {
+        if (this.stopperPrompt != null)
+            if (!uiExecContext.getPrompt().promptBoolean(this.stopperPrompt.getMessage(), true))
+                return new SimpleFailed("Aborted by the user.");
```
Comment (lincolnthree) [inline ui/addon/src/main/java/org/jboss/windup/ui/WindupCommand.java]: Why not just use Results.fail("message")?

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/config/ScanPackagesOption.java [hunk 5]
```diff
@@ -28,7 +33,7 @@ public String getName()
     @Override
     public String getLabel()
     {
-        return "Scan Java Packages";
+        return "Java packages to scan.";
```
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/config/ScanPackagesOption.java]: I don't think the others use a hard-stop at the end ("."), so won't this look out of place?

**Comment by lincolnthree** [comment]
Merged.

**Pull request closed without merging**